File: 01_Language/01_Functions/python/array_flip.py
# coding: utf-8

def array_flip(arr):
    if isinstance(arr, dict):
        return dict((v, k) for k, v in arr.items())
    else:
        return dict((v, k) for k, v in enumerate(arr))


if __name__ == '__main__':
    input = ["oranges", "apples", "pears"]
    print(array_flip(input))

    input2 = {'a': 1, 'b': 1, 'c': 2}
    print(array_flip(input2))

File: 01_Language/01_Functions/js/array_flip.js
function array_flip(arr) {
    let res = {}
    for (let i in arr) {
        res[arr[i]] = i;
    }
    return res
}

let input = ["oranges", "apples", "pears"]
console.log(array_flip(input));

let input2 = {a: 1, b: 1, c: 2}
console.log(array_flip(input2))
